### PR TITLE
Support for neovim built-in LSP client

### PR DESCRIPTION
**Pull request by pwntester**

### CODE REVIEW

File: autoload/vista/executive/nvim_lsp.vim [hunk 1]
```diff
@@ -0,0 +1,70 @@
+" Copyright (c) 2019 Alvaro Muñoz
+" MIT License
+" vim: ts=2 sw=2 sts=2 et
+
+let s:provider = fnamemodify(expand('<sfile>'), ':t:r')
+
+let g:vista_executive_nvim_lsp_reload_only = v:false
```
Comment (liuchengxu) [inline autoload/vista/executive/nvim_lsp.vim]: I'm guessing changing to the global variable is due to neovim lua does not have `vim.s.foo` support?

Comment (pwntester) [reply autoload/vista/executive/nvim_lsp.vim]: correct, we can access ot for read but not change them from lua closure

File: autoload/vista/executive/nvim_lsp.vim [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  endwhile
+  return get(s:, 'data', {})
+endfunction
+
+function! vista#executive#nvim_lsp#SetData(data) abort
+  let s:data = a:data
+endfunction
+
+function! s:RunAsync() abort
+    call vista#SetProvider(s:provider)
```
Comment (liuchengxu) [inline autoload/vista/executive/nvim_lsp.vim]: Is there any way to check if the built-in LSP support of neovim is avaliable?

Comment (pwntester) [reply autoload/vista/executive/nvim_lsp.vim]: Sure, I have added a check for nvim-0.5 to make sure both neovim and the minimum right version are used

**Comment by liuchengxu** [comment]
Overall looks good even though I actually have not tried it. @weilbith I believe you could try this PR :(.

**Comment by liuchengxu** [comment]
Merged, thanks!

**Comment by weilbith** [comment]
Thanks for the implementation! :upside_down_face:

**Pull request merged**